scr/src/main/java/org/apache/felix/scr/impl/runtime/ServiceComponentRuntimeImpl.java

```diff
@@ -117,6 +117,10 @@ public Collection<ComponentConfigurationDTO> getComponentConfigurationDTOs(Compo
         try
         {
             ComponentHolder<?> holder = getHolderFromDescription( description);
+            // the holder can also be null if the associated component is deregistered
```

Comment:
If null is allowed to be returned here then I think we need null checks in all 4 locations it is called:

`disableComponent(ComponentDescriptionDTO)`
`enableComponent(ComponentDescriptionDTO)`
`getComponentConfigurationDTOs(ComponentDescriptionDTO)`
`isComponentEnabled(ComponentDescriptionDTO)`

Also, looking at `getHolderFromDescription(ComponentDescriptionDTO description)` I think it has an issue where it could get called on an uninstalled bundle in which case `Bundle b = context.getBundle(bundleId);` will result in a `null` value in `b` causing an NPE also.